### PR TITLE
Update Packeges

### PR DESCRIPTION
**Pull request by leonardofrp5**
## Summary

Update:
- react-router
- react-router-dom
- testing-library
- react-hook-form

### CODE REVIEW

File: generators/app/templates/src/components/Routes/index.tsx [hunk 1]
```diff
@@ -15,11 +14,16 @@ function Routes() {
...
-              <RouteItem key={path} path={path} redirectTo={redirectTo?.(user)} {...config} />
+        <RoutesSwitch>
+            {ROUTES.map(({ redirectTo, path, element, ...config }) => (
+              <Route
+                key={path}
+                path={path}
+                element={redirectTo?.(user) ? <Navigate to="/logout" /> : element}
+                {...config}
+              />
             ))}
```
Comment (DavidRojas1612) [inline generators/app/templates/src/components/Routes/index.tsx]:
```suggestion
         <Route path="/" element={<Dashboard />}>
            {ROUTES.map(({ redirectTo, path, element, ...config }) => (
              <Route
                key={path}
                path={path}
                element={redirectTo?.(user) ? <Navigate to="/logout" /> : element}
                {...config}
              />
            ))}
       </Route>    
```

File: generators/app/templates/src/components/Routes/index.tsx [hunk 2]
```diff
@@ -1,11 +1,10 @@
-import { BrowserRouter as Router, Switch } from 'react-router-dom';
+import { BrowserRouter as Router, Routes as RoutesSwitch, Route, Navigate } from 'react-router-dom';
```
Comment (DavidRojas1612) [inline generators/app/templates/src/components/Routes/index.tsx:2]: import screen dashboard here

File: generators/app/templates/src/screens/Dashboard/index.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Switch } from 'react-router-dom';
+import { Routes } from 'react-router-dom';
```
Comment (DavidRojas1612) [inline generators/app/templates/src/screens/Dashboard/index.tsx]:
```suggestion
import { Outlet } from 'react-router-dom';
```

File: generators/app/templates/src/screens/Dashboard/index.tsx [hunk 4]
```diff
@@ -7,10 +7,11 @@ import Home from './screens/Home';
 
 function Dashboard() {
   return (
-    <Switch>
-      <RouteItem exact path={PATHS.home} component={Home} />
-    </Switch>
+    <Routes>
+      <RouteItem path={PATHS.home} element={<Home />} />
+    </Routes>
```
Comment (DavidRojas1612) [inline generators/app/templates/src/screens/Dashboard/index.tsx]:
```suggestion
    <div>
      <h1> home root aplication </h1>
      <Outlet />
    </div>
```

**Review by DavidRojas1612 — CHANGES_REQUESTED** [review]
Great job!!! please check some comments i put for you

**Review by DavidRojas1612 — APPROVED** [review]

File: generators/app/templates/src/components/Routes/index.tsx [hunk 5]
```diff
@@ -15,11 +15,18 @@ function Routes() {
...
-          <Switch>
-            {ROUTES.map(({ redirectTo, path, ...config }) => (
-              <RouteItem key={path} path={path} redirectTo={redirectTo?.(user)} {...config} />
+        <RoutesSwitch>
+        <Route path="/" element={<Dashboard />}>
+            {ROUTES.map(({ redirectTo, path, element, ...config }) => (
+              <Route
+                key={path}
+                path={path}
+                element={redirectTo?.(user) ? <Navigate to="/logout" /> : element}
```
Comment (pabloferro) [inline generators/app/templates/src/components/Routes/index.tsx]: Why did you moved the redirectTo logic here?
Also `redirectTo` returns a string with the route, and now your are navigating to `/logout` always.

**Review by pabloferro — CHANGES_REQUESTED** [review]

File: generators/app/templates/src/components/Routes/index.tsx (hunk 5 above)
Comment (leonardofrp5) [reply generators/app/templates/src/components/Routes/index.tsx]: what do you think

File: generators/app/templates/src/components/Routes/index.tsx [hunk 6]
```diff
@@ -15,11 +15,18 @@ function Routes() {
...
-          <Switch>
-            {ROUTES.map(({ redirectTo, path, ...config }) => (
-              <RouteItem key={path} path={path} redirectTo={redirectTo?.(user)} {...config} />
+        <RoutesSwitch>
+        <Route path="/" element={<Dashboard />}>
+            {ROUTES.map(({ redirectTo, path, element, ...config }) => (
+              <Route
+                key={path}
+                path={path}
+                element={redirectTo?.(user) ? <Navigate to={redirectTo?.(user) as any} /> : element}
```
Comment (pabloferro) [inline generators/app/templates/src/components/Routes/index.tsx]: Could this be `as string` instead?

File: generators/app/templates/src/components/Routes/constants.tsx [hunk 7]
```diff
@@ -5,28 +5,27 @@ import { User } from 'contexts/UserContext/reducer';
...
 /* When adding routes, add them ABOVE the Home route
  * or it will redirect you to incorrect screens */
 export const ROUTES = [
   // Leaving this as an example for then the Login screen exists
   {
-    exact: false,
-    path: PATHS.login,
-    component: () => <span>login</span>,
+    path: PATHS.logout,
+    element: <Logout />,
```
Comment (pabloferro) [inline generators/app/templates/src/components/Routes/constants.tsx]: I think you should revert this to `Login`

Comment (leonardofrp5) [reply generators/app/templates/src/components/Routes/constants.tsx:21]: The element property now receives a component and not a string

**Review by pabloferro — APPROVED** [review]

**Pull request merged**